src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -411,6 +411,23 @@ internal static void ConfigureApplicationInsights(HostBuilderContext context, IL
                         o.EnableDependencyTracking = false;
                     });
                 }
+
+                if (SystemEnvironment.Instance.IsApplicationInsightsAgentEnabled())
+                {
+                    // Skip sending user generated logs to AI and QuickPulse if worker AI agent is configured, worker will send these logs to AI and Quickpulse service.
+                    builder.Services.Configure<LoggerFilterOptions>(options => options.AddFilter<ApplicationInsightsLoggerProvider>((category, logLevel) =>
+                    {
+                        // match Function.<FunctionName>.User
+                        if (category.Length > 14 && category.EndsWith(".User", StringComparison.Ordinal))
```

Comment:
Benchmarked hashset and regex as well, it seems EndsWith is very efficient in this simple scenario.